src/chatgpt.rs

```diff
@@ -98,8 +124,92 @@ impl fmt::Display for Role {
     }
 }
 
+fn ensure_config_file(dir: &PathBuf) -> std::io::Result<PathBuf> {
```

Comment:
♻️ I think this should move into utils, since it isn't really related to the chatgpt client, and could be used in any scenario.